onnxruntime/core/session/IOBinding.cc

```diff
@@ -11,38 +11,34 @@ namespace onnxruntime {
 IOBinding::IOBinding(const SessionState& session_state) : session_state_(session_state) {
 }
 
-static std::pair<bool, size_t> Contains(const std::vector<std::string>& names, const std::string& name) {
-  auto it = std::find(std::begin(names), std::end(names), name);
-  if (it == std::end(names)) {
-    return {false, 0};
-  }
-  return {true, it - std::begin(names)};
-}
-
 common::Status IOBinding::BindInput(const std::string& name, const OrtValue& ml_value) {
-  auto rc = Contains(feed_names_, name);
-
-  auto add_or_replace = [this, &name](const bool exists, size_t index, const OrtValue& value) {
-    if (exists) {
-      feeds_[index] = value;
-    } else {
-      feed_names_.push_back(name);
-      feeds_.push_back(value);
-    }
-  };
+  ORT_ENFORCE(mapped_feed_names_.size() == feed_names_.size(), "Size mismatch(0):", mapped_feed_names_.size(), "!=", feed_names_.size());
+  auto it = mapped_feed_names_.emplace(name, feed_names_.size());
+  size_t index = it.first->second;
+  if (it.second) {
+    feed_names_.push_back(name);
+    OrtValue new_mlvalue;
+    feeds_.push_back(new_mlvalue);
+    ORT_ENFORCE(mapped_feed_names_.size() == feed_names_.size(), "Size mismatch(1):", mapped_feed_names_.size(), "!=", feed_names_.size(), " index=", index, " it.second=", it.second);
+    ORT_ENFORCE(mapped_feed_names_.size() == feeds_.size(), "Size mismatch(2):", mapped_feed_names_.size(), "!=", feeds_.size(), " index=", index, " it.second=", it.second);
+  }
+  ORT_ENFORCE(mapped_feed_names_.size() == feed_names_.size(), "Size mismatch(3):", mapped_feed_names_.size(), "!=", feed_names_.size(), " index=", index, " it.second=", it.second);
 
   if (ml_value.IsTensor() || ml_value.IsSparseTensor()) {
     OrtValue new_mlvalue;
+    // Do not replace new_mlvalue by feeds_[index] in the following line.
+    // It may copy the data instead of copying the pointer.
```

Comment:
Would you please elaborate on this comment? I am not sure I understood the context entirely.